hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryWorkflowImpl.java

```diff
@@ -157,7 +163,7 @@ public void handleQuery(
         // 1. Parse and check header
         final Query query;
         try {
-            query = Query.PROTOBUF.parse(requestBuffer.toReadableSequentialData());
+            query = queryParser.parse(requestBuffer.toReadableSequentialData());
```

Comment:
I believe, we need to parse everything in strict mode in the app.

Reply:
done!